ceph-releases/kraken/ubuntu/16.04/daemon/start_mon.sh

```diff
@@ -1,6 +1,74 @@
 #!/bin/bash
 set -e
 
+IPV4_REGEXP='[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}\.[0-9]\{1,3\}'
+IPV4_NETWORK_REGEXP="$IPV4_REGEXP/[0-9]\{1,2\}"
+
+function flat_to_ipv6 {
+  # Get a flat input like fe800000000000000042acfffe110003 and output fe80::0042:acff:fe11:0003
+  # This input usually comes from the ipv6_route or if_inet6 files from /proc
+
+  # First, split the string in set of 4 bytes with ":" as separator
+  value=$(echo "$@" | sed -e 's/.\{4\}/&:/g' -e '$s/\:$//')
+
+  # Let's remove the useless 0000 and "::"
+  value=${value//0000/:};
+  while $(echo $value | grep -q ":::"); do
+    value=${value//::/:};
+  done
+  echo $value
+}
+
+function get_ip {
+  NIC=$1
+  # IPv4 is the default unless we specify it
+  IP_VERSION=${2:-4}
+  # We should avoid reporting any IPV6 "scope local" interface that would make the ceph bind() call to fail
+  if command -v ip &>/dev/null; then
+    ip -$IP_VERSION -o a s $NIC | grep "scope global" | awk '{ sub ("/..", "", $4); print $4 }' || true
+  else
+    case "$IP_VERSION" in
+      6)
+        # We don't want local scope, so let's remove field 4 if not 00
+        ip=$(flat_to_ipv6 $(grep $NIC /proc/net/if_inet6 | awk '$4==00 {print $1}'))
+        # IPv6 IPs should be surrounded by brackets to let ceph-monmap being happy
+        echo "[$ip]"
+        ;;
+      *)
+        grep -o "$IPV4_REGEXP" /proc/net/fib_trie | grep -vEw "^127|255$|0$" | head -1
+        ;;
+    esac
+  fi
+}
+
+function get_network {
+  NIC=$1
+  # IPv4 is the default unless we specify it
+  IP_VERSION=${2:-4}
+
+  case "$IP_VERSION" in
+    6)
+      if command -v ip &>/dev/null; then
```

Comment:
since we test the existence of that command, shouldn't we have a function for this?